cloudflare/resource_cloudflare_api_token_test.go

```diff
@@ -42,6 +42,29 @@ func TestAccAPIToken(t *testing.T) {
 	})
 }
 
+func TestAccCloudflareAPITokenMultiplePermissionsGroups(t *testing.T) {
+	// Temporarily unset CLOUDFLARE_API_TOKEN if it is set as the API token
+	// endpoint does not yet support the API tokens without an explicit scope.
+	if os.Getenv("CLOUDFLARE_API_TOKEN") != "" {
+		defer func(apiToken string) {
+			os.Setenv("CLOUDFLARE_API_TOKEN", apiToken)
+		}(os.Getenv("CLOUDFLARE_API_TOKEN"))
+		os.Setenv("CLOUDFLARE_API_TOKEN", "")
+	}
+
+	rnd := generateRandomResourceName()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
```

Comment:
as we're now needing a schema migration we'll need to extend this a little.

the first is that we'll need to test the data migration from a list => set. https://www.terraform.io/docs/extend/resources/state-migration.html covers some examples of asserting that.

the second is that we'll want to update this test to be 2 steps. the first will create the resource with the permissions and the second will be the same permissions but in a different order and we'll want to assert there is no difference which will ensure we're ignoring the ordering.

Reply:
I added a test which fails (1st commit) and then a fix for that (2nd commit).

But yeah, I get  your point. I can add another step with different order, but don't know how to assert there is no difference. Is there some example of something similar?